src/common/kernel.cpp

```diff
@@ -55,6 +55,67 @@
 #endif
 #endif
 
+#ifdef TRACY_ENABLE
+void* operator new(std::size_t count)
+{
+    void* ptr = malloc(count);
+    TracyAlloc(ptr, count);
+    return ptr;
+}
+
+void operator delete(void* ptr) noexcept
+{
+    TracyFree(ptr);
+    free(ptr);
+    ptr = nullptr;
```

Comment:
Assigning to `ptr` here does not change the actual value of pointer outside this scope so `ptr = nullptr` has no effect outside (tested on clang 14.0.0).

Reply:
Bah humbug, you're quite right, I'll have to go with `destroy()` in place of the actual calls to delete, and set up some CI rules to keep an eye out for calls to delete. Thanks for the heads up 👍 

```cpp
template <typename T>
inline void destroy(T*& ptr)
{
    delete ptr;
    ptr = nullptr;
}
```